zulip_bots/zulip_bots/lib.py

```diff
@@ -47,12 +49,13 @@ def show_error_and_exit(self):
         sys.exit(1)
 
 class ExternalBotHandler(object):
-    def __init__(self, client):
-        # type: (Client) -> None
+    def __init__(self, client, bot_name):
+        # type: (Client, string) -> None
         # Only expose a subset of our Client's functionality
         user_profile = client.get_profile()
         self._rate_limit = RateLimit(20, 5)
         self._client = client
+        self._bot_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'bots', bot_name)
```

Comment:
I think we want the library to be as flexible as possible.  Nothing prevents us from writing more flexible code in `server.py` down the road, or, if the server is liberal about what it runs, we can have an option where the `server` doesn't pass in the `root_dir`.  In the library, I think if `root_dir` is `None`, we assume it doesn't matter where the files come from.